### PR TITLE
Decrement ECS subnet count to 4

### PR DESCRIPTION
**Pull request by jjtimmons**
- reduction in total subnet count to 4
- in a few places I was getting duplicate resource exceptions: expanded the places we use `${local.secret_prefix}`

#### Testing

`tf apply`

Screen Shot 2022-03-24 at 5 02 16 PM

### CODE REVIEW

File: modules/hcp-ecs-client/services.tf [hunk 1]
```diff
@@ -20,7 +20,7 @@ module "acl-controller" {
 }
 
 resource "aws_iam_role" "frontend-task-role" {
-  name = "frontend_${local.scope}_task_role"
+  name = "${local.secret_prefix}_frontend_${local.scope}_task_role"
```
Comment (i0rek) [inline modules/hcp-ecs-client/services.tf:23]: isn't `local.scope` the same like `local.secret_prefix`? Not literally the same, but also something random?

File: modules/hcp-ecs-client/services.tf [hunk 2]
```diff
@@ -33,10 +33,12 @@ resource "aws_iam_role" "frontend-task-role" {
       }
     ]
   })
+
+  name_prefix = local.secret_prefix
```
Comment (i0rek) [inline modules/hcp-ecs-client/services.tf:37]:
```suggestion
  name_prefix = local.scope
```

I would like to use `scope` for these things. `secret_prefix` should only be used for secret prefixes for the acl controller.

**Review by i0rek — APPROVED** [review]
Left a couple of comments, the change itself looks good!

File: modules/hcp-ecs-client/services.tf (hunk 1 above)
Comment (jjtimmons) [reply modules/hcp-ecs-client/services.tf:23]: Ah yeah, that's right, my bad

Looking at this again, I was actually hitting the duplicate name down in `mesh-task/iam/`:

```
// Create the execution role and attach policies
resource "aws_iam_role" "execution" {
  count = local.create_execution_role ? 1 : 0
  name  = "${var.family}-execution"
```

Where family name was not unique so the role wasn't

**Pull request merged**